examples/acl/src/backend/app/controllers/posts.clj

```diff
@@ -0,0 +1,193 @@
+(ns controllers.posts
+  (:require
+    [clojure.string :as str]
+    [clojure.walk :refer [keywordize-keys]]
+    [framework.acl.core :as acl]
+    [honeysql.core :as sql]
+    [honeysql.helpers :refer :all :as helpers]
+    [next.jdbc :as jdbc]
+    [ring.middleware.params :as par]
+    [views.posts]
+    [xiana.core :as xiana])
+  (:import
+    (java.util
+      UUID)))
+
+(defn unauthorized
+  [state]
+  (xiana/error (assoc state :response {:status 401 :body "Unauthorized"})))
+
+(defn guest
+  [state]
+  (xiana/ok (assoc-in state [:session :user] {:id (UUID/randomUUID)
+                                              :role :guest})))
+
+(defn require-logged-in
+  "Tricky login, session should handle user data"
```

Comment:
if you need session, and we have a session component, then use it
Otherwise this example presents an intent: Despite we have a component for it in the framework, rather write your own